### DISCUSSION TITLE
Unsetting a configuration doesn't work as expected

### DISCUSSION THREAD

**Discussion by railken** [laravel/framework#34522] [General]
- Laravel Version: 6.18.40, 7.28.3, 8.6.0
- PHP Version: 7.2, 7.3, 7.4
- Database Driver & Version: mysql 5.7

### Description:

After unsetting a config the default value doesn't work anymore

I think this happens because instead of unsetting the value with key, the value is set to null [here](https://github.com/laravel/framework/blob/8.x/src/Illuminate/Config/Repository.php#L177)

### Steps To Reproduce:

```php
$cfg = $this->app->get('config');
$this->assertEquals(3, $cfg->get('foo', 3));

$cfg->set('foo', 5);
$this->assertEquals(5, $cfg->get('foo'));

$cfg->offsetUnset('foo');
$this->assertEquals(7, $cfg->get('foo', 7)); // fails, $cfg->get('foo')  return null instead of 7
```

### Tests
Result of test: [link](https://github.com/railken/laravel-bug-report-config/runs/1161451748#step:8:28)

**Comment by driesvints** [comment]
Hi there,

Thanks for reporting but it looks like this is a question which can be asked on a support channel. Please only use this issue tracker for reporting bugs with the library itself. 

You can post support questions on our discussions tab instead so I'm moving it over there. I won't be able to help you out personally but others who are watching this repository can help you out.

Alternatively, you can post this question on one of the following channels:

- [Laracasts Forums](https://laracasts.com/discuss)
- [Laravel.io Forums](https://laravel.io/forum)
- [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
- [Discord](https://discordapp.com/invite/KxwQuKb)
- [Larachat](https://larachat.co)
- [IRC](https://webchat.freenode.net/?nick=laravelnewbie&channels=%23laravel&prompt=1)

Thanks.

**Reply by railken** [reply]
@driesvints Hi, this is not a support request. 

Either is a bug or an intended behavior. 

Please tell me why this has been moved to discussions. 

Thanks

**Comment by lagbox** [comment]
You are correct in your conclusion of why that is the case. It doesn't unset the key, it just sets it to `null`. The `get` uses `Arr::get` which uses `Arr::exists` to see if the key exists. Since the key exists it returns the value which is `null`.

**Reply by railken** [reply]
I thought the issue was deleted, thanks to you i now know the existence of `transfer to discussion`.
